beluga_tutorial/src/visualization.py

```diff
@@ -0,0 +1,164 @@
+    ax_landmark.set_xlabel("States")
+    ax_landmark.set_xlim(-5, 100)
+    ax_landmark.set_ylim(0, 2)
+    ax_landmark.set_xticks(np.arange(-5, 101, 5))
+
+    ground_truth = yaml_data['simulation_records'][index]["ground_truth"]
+    ax_ground_truth = axs[5]
+    ax_ground_truth.clear()
+    ax_ground_truth.bar(ground_truth, 1, color='green')
+    ax_ground_truth.set_title(f"Ground Truth: {ground_truth}")
+    ax_ground_truth.set_xlabel("States")
+    ax_ground_truth.set_xlim(-5, 100)
+    ax_ground_truth.set_ylim(0, 2)
+    ax_ground_truth.set_xticks(np.arange(-5, 101, 5))
+
+    mean = "{:.3f}".format(yaml_data['simulation_records'][index]["estimation"]["mean"])
+    sd = "{:.3f}".format(yaml_data['simulation_records'][index]["estimation"]["sd"])
+    plt.text(
+        0.5,
+        0.5,
+        f"Mean: {mean}\nSD: {sd}",
+        ha='center',
+        va='center',
+        transform=axs[5].transAxes,
+        bbox=dict(facecolor='white', alpha=0.5),
+    )
+
+    plt.tight_layout()
+    plt.draw()
+
+
+def on_key(event, yaml_data, axs, current_frame_container, num_frames):
+    if event.key == 'right':
+        current_frame_container[0] = (current_frame_container[0] + 1) % num_frames
+    elif event.key == 'left':
+        current_frame_container[0] = (current_frame_container[0] - 1) % num_frames
+    plot_stages(yaml_data, axs, current_frame_container[0])
+
+
+def main(argv=None) -> int:
+    parser = argparse.ArgumentParser(
+        description="This script uses the matplotlib to visualize the data generated\
+            by beluga_tutorial example code."
+    )
+
+    parser.add_argument(
+        '-p',
+        '--record-file-path',
+        type=str,
+        help='Absolute path to the record file generated by the beluga_tutorial example code',
+        required=True,
+    )
+
+    parser.add_argument(
+        '-m',
+        '--manual-control',
+        type=bool,
+        help='Manual controlling the time steps of the visualization',
+        required=False,
+        default=False,
+    )
+
+    parser.add_argument(
+        '-i',
+        '--interval-ms',
+        type=int,
+        help='Delay between frames in milliseconds.',
+        required=False,
+        default=250,
+    )
+
+    args = parser.parse_args(argv)
+
+    yaml_data = read_yaml_file(args.record_file_path)
+
+    if yaml_data:
+        fig, axs = plt.subplots(6, 1)
+        num_frames = len(yaml_data['simulation_records'])
+
+        if args.manual_control:
+            current_frame_container = [0]
+            plot_stages(yaml_data, axs, current_frame_container[0])
+            fig.canvas.mpl_connect(
+                'key_press_event',
+                lambda event: on_key(
+                    event, yaml_data, axs, current_frame_container, num_frames
+                ),
+            )
+        else:
+
+            def update(current_frame):
+                plot_stages(yaml_data, axs, current_frame)
+
+            animation.FuncAnimation(
```

Comment:
The animation does not seem to work in my machine.

Reply:
Yes, now in doesn't work in my machine also, there is a problem because `animation.FuncAnimation()` function throwing me the following warning: `UserWarning: Animation was deleted without rendering anything. This is most likely not intended. To prevent deletion, assign the Animation to a variable, e.g. `anim`, that exists until you output the Animation using plt.show() or anim.save().`
At the same time, the linter or the pre-commit doens't let me commit a code with unused variables. Maybe I can to use some type of indicator that `anim` is an unused variable.

Reply:
Linters usually have a way of preventing them from changing certain parts of the code you know are correct.
Which one of the [linters](https://github.com/Ekumen-OS/beluga/blob/main/.pre-commit-config.yaml) is preventing you from adding that line?

Reply:
this linter `flake8`. I'm trying to do the following but the linter doesn't like the `_anim` (I added the underscore because I saw this in some place but apparently it doesn't work)
```python
_anim = animation.FuncAnimation(
            fig,
            update,
            frames=num_frames,
            blit=False,
            interval=args.interval_ms,
            repeat=False,
        )
```

Reply:
This might be useful:
https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors